### PR TITLE
Replace typedefs in core/src

### PR DESCRIPTION
**Pull request by masterleinad**
Part of #3213.

### CODE REVIEW

File: core/src/Cuda/Kokkos_Cuda_Instance.hpp [hunk 1]
```diff
@@ -35,8 +35,8 @@ struct CudaTraits {
     KernelArgumentLimit = 0x001000 /*  4k bytes */
   };
 
-  typedef unsigned long
-      ConstantGlobalBufferType[ConstantMemoryUsage / sizeof(unsigned long)];
+  using ConstantGlobalBufferType =
+      long[ConstantMemoryUsage / sizeof(unsigned long)];
```
Comment (jeffmiles63) [inline core/src/Cuda/Kokkos_Cuda_Instance.hpp:39]: does it matter that this changed from an unsigned long to a long?

Comment (masterleinad) [reply core/src/Cuda/Kokkos_Cuda_Instance.hpp:39]: That should not have changed. Thanks for pointing out!

Comment (masterleinad) [reply core/src/Cuda/Kokkos_Cuda_Instance.hpp]: Fixed.

**Review by jeffmiles63 — APPROVED** [review]
Looks good

**Pull request merged**